src/nbson.erl

```diff
@@ -43,14 +43,27 @@ decode(Data) ->
     Path :: document_path() | nbson_key(),
     Document :: document(),
     Result :: nbson_value().
-get([], Document) ->
+get(Path, Document) ->
+    do_get(Path, Document).
```

Comment:
Why we don't move the implementation of do_get here? get is a function that just calls a function.

Reply:
That was the original approach, however `gradualizer` found a type error.
The type annotation for the second argument of `get`, namely `Document`, is `document()`. However, given the actual implementation of this function, we usually end up calling the function with a non-`document()` value as an argument (i.e., we use the second argument as the accumulator in a recursive flow, so the final value can be from a different type).